src/libFLAC/CMakeLists.txt

```diff
@@ -21,6 +21,9 @@ if(FLAC__CPU_X86_64 OR FLAC__CPU_IA32)
     if(WITH_AVX AND MSVC)
         set_source_files_properties(fixed_intrin_avx2.c lpc_intrin_avx2.c stream_encoder_intrin_avx2.c lpc_intrin_fma.c PROPERTIES COMPILE_FLAGS /arch:AVX2)
     endif()
+    if(WITH_AVX AND WIN32 AND NOT MSVC)
+        set_source_files_properties(fixed_intrin_avx2.c lpc_intrin_avx2.c stream_encoder_intrin_avx2.c lpc_intrin_fma.c PROPERTIES COMPILE_FLAGS -mavx2)
```

Comment:
I don't understand why this is necessary. I regularly build with Clang on Win32 through MinGW, and that works perfectly fine without this patch.

So, either the version of Clang you're using is modified in some way, or there is something with the MSVC header files happening here.

In any case, the `if()` is too broad because it also applies to MinGW.

Reply:
This still doesn't make any sense to me. Why would this only be necessary on Clang on Windows with MSVC headers? I'd rather understand this first. It seems like some logic is missing in https://github.com/xiph/flac/blob/master/src/libFLAC/include/private/cpu.h

Reply:
Apparently this is a bug in ClangCL: https://github.com/llvm/llvm-project/issues/53520